flatsurf/geometry/surface_objects.py

```diff
@@ -1144,6 +1144,72 @@ def _test_inverse(self, **options):
             check=True,
         )
 
+    def _homology_(self, H):
+        r"""
+        Return this saddle connection as a chain of edges in the homology group ``H``.
+
+        EXAMPLES::
+
+            sage: from flatsurf import *
+            sage: S = translation_surfaces.mcmullen_L(1,1,1,1)
+            sage: H = S.homology()
+            sage: holonomy = lambda x: sum(coeff * S.polygon(label).edge(e) for (label, e), coeff in dict(x._chain).items())
+            sage: for sc in S.saddle_connections(5):
+            ....:     h = H(sc)
+            ....:     hol1 = holonomy(h)
+            ....:     hol2 = sc.holonomy()
+            ....:     assert hol1 == hol2, (sc, h, hol1, hol2)
+        """
+        from .homology import SimplicialHomologyGroup
+
+        if not isinstance(H, SimplicialHomologyGroup):
+            raise TypeError("H must be a SimplicialHomologyGroup")
+
+        surface = self._surface
+        if H._surface != surface:
+            raise ValueError("homology and surface do not match")
+
+        traj = self.start_tangent_vector().straight_line_trajectory()
+        n = 10
+        while not traj.is_saddle_connection():
+            traj.flow(2**n)
+            n += 1
+        segments = traj.segments()
+
+        if len(segments) == 1 and segments[0].is_edge():
+            label = segments[0].polygon_label()
+            e = segments[0].edge()
+            return H((label, e))
```

Comment:
Is this necessary or just an optimization?

Reply:
It is necessary : the segment has a start in the polygon on the left side and the end on the right side. The generic below does not work.

Reply:
Here is an example
```
sage: S = translation_surfaces.mcmullen_L(1,1,1,1)
sage: traj = S.tangent_vector(0, (0, 0), (1, 0)).straight_line_trajectory()
sage: traj.flow(1)
sage: s = traj.segments()[0]
sage: s
Segment in polygon 0 starting at (0, 0) and ending at (1, 1)
```
Let me notice that the string representation of the segment is quite confusing : it would better have been `ending at (1,0)`. The reason is that (0, 0) refers to polygon 0 while (1, 1) refers to polygon 1. This becomes apparent when looking at the start and end
```
sage: s.start()
SimilaritySurfaceTangentVector in polygon 0 based at (0, 0) with vector (1, 0)
sage: s.end()
SimilaritySurfaceTangentVector in polygon 1 based at (1, 1) with vector (-1, 0)
```

Reply:
I opened #309